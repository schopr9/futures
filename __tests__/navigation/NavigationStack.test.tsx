import React from 'react';
import { render } from '@testing-library/react-native';
import App from 'app/navigation/NavigationStack';

import * as redux from 'react-redux';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

describe('NavigationStack', () => {
  it('mounts Maintenance route if maintenance mode is enabled', () => {
    const useDispatchSpy = jest.spyOn(redux, 'useDispatch');
    const mockDispatchFn = jest.fn();
    useDispatchSpy.mockReturnValue(mockDispatchFn);
    const { toJSON } = render(
      <App theme={{ dark: true, colors: { primary: 'blue' } }} />,
    );
    expect(toJSON).toMatchSnapshot();
  });
});
