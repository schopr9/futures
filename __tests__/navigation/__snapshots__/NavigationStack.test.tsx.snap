// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationStack mounts Maintenance route if maintenance mode is enabled 1`] = `[Function]`;
